### ISSUE TITLE
Inconsistent behavior when importing CategoricalAccuracy from tensorflow.keras.metrics and tensorflow.python.keras.metrics

### ISSUE THREAD

**Issue by alexandresee** [keras-team/keras#16620]
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 20.04.3 LTS (Focal Fossa)
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): v2.8.0-rc1-32-g3f878cff5b6 2.8.0
- Python version: 3.8.10
- Bazel version (if compiling from source):
- GPU model and memory: NVIDIA V100, 32GB
- Exact command to reproduce:

**Describe the problem**.

Importing CategoricalAccuracy from `tensorflow.keras.metrics` and `tensorflow.python.keras.metrics` is not equivalent. The former seems to compute categorical accuracy adequately. However, the latter doesn't. It seems to be applying a smoothing function over the results. See plots at the bottom of the [colab notebook](https://colab.research.google.com/drive/1QzMXITZD2Ebok9QoKicZPhJEHSN9x2FB?usp=sharing).

**Describe the current behavior**.

1) `tensorflow.keras.metrics` and `tensorflow.python.keras.metrics` do not provide `CategoricalAccuracy` metrics that have consistent behaviors.

2) `tensorflow.python.keras.metrics.CategoricalAccuracy` does not seem to compute categorical accuracy adequately. 

**Describe the expected behavior**.

Both `tensorflow.keras.metrics.CategoricalAccuracy` and `tensorflow.python.keras.metrics.CategoricalAccuracy` should provide the same metric, which should compute categorical accuracy properly.

- Do you want to contribute a PR? (yes/no): no

**Standalone code to reproduce the issue**.

[Colab](https://colab.research.google.com/drive/1QzMXITZD2Ebok9QoKicZPhJEHSN9x2FB?usp=sharing)

**Comment by alexandresee** [comment]
I had not found references to `tensorflow.python` being [private](https://stackoverflow.com/a/58279629) before posting this issue. Given this information, I will close the issue. However, I do find this type of inconsistency rather hard to explain. If anyone could provide any information related to why the behavior from both metrics is so different, it would be greatly appreciated.
